src/components/application_manager/test/commands/mobile/alert_request_test.cc

```diff
@@ -32,206 +32,154 @@
+  EXPECT_EQ((*vr_command_result)[am::strings::msg_params][am::strings::success]
+                .asBool(),
+            false);
+  EXPECT_EQ(
+      (*vr_command_result)[am::strings::msg_params][am::strings::result_code]
+          .asInt(),
+      static_cast<int32_t>(am::mobile_api::Result::GENERIC_ERROR));
 }
 
-TEST_F(AlertRequestTest, Run_SUCCESS) {
-  MessageSharedPtr msg(CreateMessage(smart_objects::SmartType_Map));
-  (*msg)[am::strings::msg_params][am::strings::alert_text1] = "alert_text1";
-  (*msg)[am::strings::msg_params][am::strings::alert_text2] = "alert_text2";
-  (*msg)[am::strings::msg_params][am::strings::tts_chunks][0]
-        [am::strings::text] = "alert_text2";
-
-  CommandPtr command(CreateCommand<AlertRequest>(msg));
-
-  MockAppPtr app(CreateMockApp());
-  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(app));
-  ON_CALL(*app, hmi_level())
-      .WillByDefault(Return(mobile_apis::HMILevel::HMI_BACKGROUND));
-  ON_CALL(*app, AreCommandLimitsExceeded(_, _)).WillByDefault(Return(false));
-
-  ON_CALL(app_mngr_, GetPolicyHandler())
-      .WillByDefault(ReturnRef(policy_interface_));
-  EXPECT_CALL(*MockMessageHelper::message_helper_mock(),
-              ProcessSoftButtons(_, _, _, _))
-      .WillOnce(Return(mobile_apis::Result::SUCCESS));
-
-  EXPECT_CALL(this->app_mngr_, ManageHMICommand(_))
-      .Times(AtLeast(1))
-      .WillRepeatedly(Return(true));
-
-  command->Run();
+TEST_F(AlertRequestTest, OnEvent_UI_UNSUPPORTED_RESOURCE) {
+  MessageSharedPtr msg_vr = CreateFullParamsUISO();
+  (*msg_vr)[am::strings::msg_params][am::strings::menu_params]
+           [am::hmi_request::parent_id] = 10u;
+  (*msg_vr)[am::strings::msg_params][am::strings::menu_params]
+           [am::strings::menu_name] = "menu_name";
+
+  utils::SharedPtr<AlertRequest> req_vr = CreateCommand<AlertRequest>(msg_vr);
+
+  MockAppPtr mock_app = CreateMockApp();
+  ON_CALL(app_mngr_, application(kConnectionKey))
+      .WillByDefault(Return(mock_app));
+  ON_CALL(*mock_app, app_id()).WillByDefault(Return(kConnectionKey));
+  MockHmiInterfaces hmi_interfaces;
+  ON_CALL(app_mngr_, hmi_interfaces()).WillByDefault(ReturnRef(hmi_interfaces));
+  ON_CALL(hmi_interfaces, GetInterfaceFromFunction(_))
+      .WillByDefault(
+          Return(am::HmiInterfaces::HMI_INTERFACE_BasicCommunication));
+  ON_CALL(hmi_interfaces, GetInterfaceState(_))
+      .WillByDefault(Return(am::HmiInterfaces::STATE_AVAILABLE));
+
+  MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+  (*msg)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::UNSUPPORTED_RESOURCE;
+  (*msg)[am::strings::msg_params][am::strings::cmd_id] = kCommandId;
+
+  Event event(hmi_apis::FunctionID::UI_Alert);
+  event.set_smart_object(*msg);
+
+  MockMessageHelper* mock_message_helper =
+      MockMessageHelper::message_helper_mock();
+  ON_CALL(*mock_message_helper, HMIToMobileResult(_))
+      .WillByDefault(Return(mobile_apis::Result::SUCCESS));
```

Comment:
@okozlovlux, you use `MockMessageHelper` in both test cases, can you move it to test fixture?
If you do, please don't forget to add `VerifyAndClearExpectations()`.